### PR TITLE
ActiveRecord:findByCondition alias fix

### PR DESCRIPTION
**Pull request by rikosage**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | I couldn't run it
| Fixed issues  | no

`ActiveRecord::findOne($id)` method doesn't includes table alias. If alias is defined in `find` or `init` methods, `ActiveRecord::findByCondition` returns error.

### CODE REVIEW

File: framework/db/ActiveRecord.php [hunk 1]
```diff
@@ -179,7 +179,7 @@ protected static function findByCondition($condition)
             if (isset($primaryKey[0])) {
                 $pk = $primaryKey[0];
                 if (!empty($query->join) || !empty($query->joinWith)) {
-                    $pk = static::tableName() . '.' . $pk;
+                    $pk = ($query->from ? key($query->from) : static::tableName()) . '.' . $pk;
```
Comment (cebe) [inline framework/db/ActiveRecord.php]: this is not correct, `$query->from` could be `['sometable']` where the key would be 0.
This should use similar approach as `ActiveQuery::getTableNameAndAlias()` but that method is protected right now.

Comment (rikosage) [reply framework/db/ActiveRecord.php]: Should I create new method like `getAlias()`? Or I can just change `protected` to `public`, but it's doubtful. What's better solution?

Comment (cebe) [reply framework/db/ActiveRecord.php]: ActiveQuery should not have public methods other than those building the query, so a private function for getting the alias should be used. It's okay to duplicate some code from ActiveQuery. I don't see a better way to solve it right now.

**Comment by rikosage** [comment]
I've reduced cyclomatic comlexity. 
Should I use Reflection for testing new private method?

**Comment by samdark** [comment]
Isn't it possible to test it w/o reflection?

File: framework/db/ActiveRecord.php [hunk 2]
```diff
@@ -174,25 +174,47 @@ protected static function findByCondition($condition)
...
+
+        if (!isset($primaryKey[0])) {
+            throw new InvalidConfigException('"' . get_called_class() . '" must have a primary key.');
+        }
+
+        if (empty($query->join) && empty($query->joinWith)) {
+            return $primaryKey[0];
+        }
+
+        $tableOrAlias = static::tableName();
```
Comment (GHopperMSK) [inline framework/db/ActiveRecord.php]: In this conditions `$query->from` returns or null or single-element array. That is why we can just use
```suggestion
        $tableOrAlias = (!$query->from || isset($query->from[0])) ? static::tableName() : key($query->from);
```
without foreach loop, `array_keys` and even `is_string` funtions.

**Comment by samdark** [comment]
Closed in favor of https://github.com/yiisoft/yii2/pull/17127

**Pull request closed without merging**